server/src/test/java/org/eclipse/hono/server/StandaloneCredentialsApiTest.java

```diff
@@ -0,0 +1,364 @@
+        ctx.assertNotNull(payload);
+        JsonObject firstSecret = pickFirstSecretFromPayload(ctx,payload);
+        ctx.assertNotNull(firstSecret);
+
+        LocalDateTime now = LocalDateTime.now();
+
+        ctx.assertTrue(firstSecret.containsKey(CredentialsConstants.FIELD_SECRETS_NOT_BEFORE));
+        String notBefore = firstSecret.getString(CredentialsConstants.FIELD_SECRETS_NOT_BEFORE);
+        LocalDateTime notBeforeLocalDate = LocalDateTime.parse(notBefore, DateTimeFormatter.ISO_OFFSET_DATE_TIME);
+        ctx.assertTrue(now.compareTo(notBeforeLocalDate) >= 0);
+
+        ctx.assertTrue(firstSecret.containsKey(CredentialsConstants.FIELD_SECRETS_NOT_AFTER));
+        String notAfter = firstSecret.getString(CredentialsConstants.FIELD_SECRETS_NOT_AFTER);
+        LocalDateTime notAfterLocalDate = LocalDateTime.parse(notAfter, DateTimeFormatter.ISO_OFFSET_DATE_TIME);
+        ctx.assertTrue(now.compareTo(notAfterLocalDate) <= 0);
+    }
+
+    private void checkPayloadGetCredentialsReturnsMultipleSecrets(final TestContext ctx,final JsonObject payload) {
+        ctx.assertNotNull(payload);
+
+        // secrets: first entry is expected to be valid, second entry may have timestamps not yet active (not checked), more entries may be avail
+        ctx.assertTrue(payload.containsKey(CredentialsConstants.FIELD_SECRETS));
+        JsonArray secrets = payload.getJsonArray(CredentialsConstants.FIELD_SECRETS);
+        ctx.assertNotNull(secrets);
+        ctx.assertTrue(secrets.size() > 1); // at least 2 entries to test multiple entries
+    }
+
+    private void checkPayloadGetCredentialsContainsAuthIdAndType(final TestContext ctx,final JsonObject payload) {
+        ctx.assertNotNull(payload);
+
+        ctx.assertTrue(payload.containsKey(CredentialsConstants.FIELD_AUTH_ID));
+        ctx.assertEquals(payload.getString(CredentialsConstants.FIELD_AUTH_ID), CREDENTIALS_USER);
+
+        ctx.assertTrue(payload.containsKey(CredentialsConstants.FIELD_TYPE));
+        ctx.assertEquals(payload.getString(CredentialsConstants.FIELD_TYPE),CREDENTIALS_TYPE);
+    }
+
+    private void checkPayloadGetCredentialsContainsDefaultDeviceIdAndIsEnabled(final TestContext ctx,final JsonObject payload) {
+        ctx.assertNotNull(payload);
+
+        ctx.assertTrue(payload.containsKey(CredentialsConstants.FIELD_DEVICE_ID));
+        ctx.assertEquals(payload.getString(CredentialsConstants.FIELD_DEVICE_ID), DEFAULT_DEVICE_ID);
+
+        ctx.assertTrue(payload.getBoolean(CredentialsConstants.FIELD_ENABLED));
+    }
+
+    private byte[] hashPassword(final String hashFunction,final String hashSalt,final String passwordToHash) {
+        try {
+            MessageDigest messageDigest = MessageDigest.getInstance(hashFunction);
+            messageDigest.update(hashSalt.getBytes("UTF-8"));
+            byte[] theHashedPassword = messageDigest.digest(passwordToHash.getBytes());
+            return theHashedPassword;
+        } catch (NoSuchAlgorithmException e) {
```

Comment:
I agree, made the writing of the test case slightly more complicated, but definitely pays off since the exact exception (if occurring) is shown by the test framework.